src/python/turicreate/test/test_sframe_s3.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2020 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+#
+from __future__ import print_function as _  # noqa
+from __future__ import division as _  # noqa
+from __future__ import absolute_import as _  # noqa
+from ..data_structures.sframe import SFrame
+
+import tempfile
+import os
+import math
+import shutil
+import pytest
+import boto3
+
+
+@pytest.mark.skipif(
+    os.environ.get("DTURI_ENABLE_SF_S3") is None,
+    reason="slow IO test; enabled when needed",
+)
+class TestSFrameS3(object):
+    @classmethod
+    def setup_class(self):
+        self.my_tempdir = tempfile.mkdtemp()
+        self.s3_client = boto3.client(
+            "s3",
+            endpoint_url=os.environ["TURI_S3_ENDPOINT"],
+            region_name=os.environ["TURI_S3_REGION"],
+        )
+        self.bucket = "tc_qa"
+        self.s3_prefix = "integration/manual/"
```

Comment:
I'm not sure "manual" makes sense in the prefix.

Much more importantly - if two different instances of these tests are running at the same time, they will step on each other. I suggest you generate a UUID and append that to the prefix.

Reply:
that's my s3 prefix, not the prefix on local fs.

Reply:
and the file is stored in S3, without uuid.

Reply:
Right, it looks like it always writes to the same place in S3. So if two instances of this script are running at the same time they will be interfering with each other because they're both writing to the same place.

Reply:
I believe s3 will attach version id to it. And our internal S3 won't allow us to delete folders.

Reply:
that's why I'm not going to have a UUID as a grouping folder.